src/main/java/org/hiero/sketch/dataset/ParallelDataSet.java

```diff
@@ -0,0 +1,101 @@
+package org.hiero.sketch.dataset;
+
+import org.hiero.sketch.dataset.api.*;
+import rx.Observable;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class ParallelDataSet<T> implements IDataSet<T> {
+    private final ArrayList<IDataSet<T>> children;
+
+    private ParallelDataSet(final Map<Integer, IDataSet<T>> elements) {
+        this.children = new ArrayList<IDataSet<T>>(elements.size());
+        for (final Map.Entry<Integer, IDataSet<T>> e : elements.entrySet())
+            this.children.add(e.getKey(), e.getValue());
+    }
+
+    public ParallelDataSet(final ArrayList<IDataSet<T>> children) {
+        this.children = children;
+    }
+
+    private int size() { return this.children.size(); }
+
+    @Override
+    public <S> Observable<PartialResult<IDataSet<S>>> map(final IMap<T, S> mapper) {
+        final ArrayList<Observable<Pair<Integer, PartialResult<IDataSet<S>>>>> obs =
+                new ArrayList<Observable<Pair<Integer, PartialResult<IDataSet<S>>>>>(this.size());
+        for (int i=0; i < this.size(); i++) {
+            final int finalI = i;
+            final Observable<Pair<Integer, PartialResult<IDataSet<S>>>> ci =
+                    this.children.get(i)
+                            .map(mapper)
+                            .map(e -> new Pair<Integer, PartialResult<IDataSet<S>>>(finalI, e));
+            obs.add(i, ci);
+        }
+        final Observable<Pair<Integer, PartialResult<IDataSet<S>>>> merged =
+                Observable.merge(obs);
+
+        final Observable<PartialResult<IDataSet<S>>> map = merged.filter(p -> p.second.deltaValue != null)
```

Comment:
Name this variable something other than map. The method name is already 'map', and there's also several invocations of map on the child IDataSets and observables.